src/QsCompiler/Tests.Compiler/TestCases/LinkingTests/Monomorphization.qs

```diff
@@ -34,6 +34,39 @@ namespace Microsoft.Quantum.Testing.Monomorphization {
 
 // =================================
 
+namespace Microsoft.Quantum.Testing.Monomorphization {
+    open Microsoft.Quantum.Testing.Generics;
+
+    @ EntryPoint()
+    operation Test4() : Unit {
+        Test4Main();
+    }
+}
+
+// =================================
+
+namespace Microsoft.Quantum.Testing.Monomorphization {
+    internal newtype MyInternalUDT = Int;
+    newtype MyPublicUDT = String;
+    
+    // The first (Internal|Public) describes the access of the operation.
+    // The second (Internal|Public) describes the type that will be passed in.
+    internal operation InternalInternalFoo<'A>(x : 'A) : Unit { }
+    internal operation InternalPublicFoo<'A>(x : 'A) : Unit { }
+    operation PublicInternalFoo<'A>(x : 'A) : Unit { }
+    operation PublicPublicFoo<'A>(x : 'A) : Unit { }
+
+    @ EntryPoint()
+    operation TestAccessModifiers() : Unit {
```

Comment:
It would be good to also add a test for when the internal type is 
- in the return type
- in the signature of an argument that is passed
- in the signature of a callable that is returned
- the array base type (argument and return)
- the type of a tuple item (argument and return)

Reply:
I've added these tests.